openmdao/recorders/sqlite_reader.py

```diff
@@ -26,6 +29,74 @@
 from io import TextIOBase
 
 
+class _InstanceWithNoClassDefinition:
+    """
+    A class used by _RestrictedUnpickler.
+
+    Used to indicate the unpickler can't generate an instance of a class
+    whose class definition is not available
+    """
+
+    pass
+
+
+class _RestrictedUnpicklerForCaseReader(pickle.Unpickler):
+
+    def __init__(self, file, *, fix_imports=True, encoding="ASCII",
+                 errors="strict", buffers=None):
+        super().__init__(file, fix_imports=fix_imports, encoding=encoding,
+                         errors=errors, buffers=buffers)
+        self.error_strings = ''  # Used to document which classes are not available
+
+    def find_class(self, module, name):
+        # Only allow classes from 'module'
+
+        module_available = importlib.util.find_spec(module) is not None
+        if module_available:  # module is available so do as normal
+            return super().find_class(module, name)
+
+        # If module not available...
+        # Save into the error_strings variable what was not available.
+        if self.error_strings:
+            self.error_strings += ', '
+        self.error_strings += f"global '{module}.{name}' is not available."
+
+        # Returning this acts as a kind of flag to indicate that
+        # the unpickler can't generate instances of classes whose class definition
+        # is not available
+        return _InstanceWithNoClassDefinition
+
+    def loads_and_return_errors(self):
+        unpickled_contents = self.load()
+        return unpickled_contents, self.error_strings
+
+
+def _loads_and_return_errors(s):
+    """
+    Unpickle input and also note errors. Analogous to pickle.loads().
+
+    But handles unpickling objects with no class definition available.
+    """
+    i = io.BytesIO(s)
+
+    # returns a tuple of the value and also error strings
+    dictionary, error_string = _RestrictedUnpicklerForCaseReader(i).loads_and_return_errors()
+
+    # remove any InstanceWithNoClassDefinition classes
+    # Only want to return legit values in the OptionsDictionary
+    keys = [key for key, value in dictionary.items()]
+    for key in keys:
+        try:
+            isInstanceWithNoClassDefinition = isinstance(dictionary[key],
+                                                         _InstanceWithNoClassDefinition)
+            if isInstanceWithNoClassDefinition:
+                dictionary.undeclare(key)
```

Comment:
Do we really want to remove the option completely?  

Seems like we would want it to show up as `Unknown Type` or something like that...  
Do things blow up if you just leave the `_InstanceWithNoClassDefinition` in there?

Reply:
At first I had it that way. Things don't blow up if you just leave the _InstanceWithNoClassDefinition in there. 

But I thought users would be confused by seeing an instance of some weird class that they didn't know anything about. On the other hand, seeing it might make them realize that something is not quite right. We do issue a warning but they might miss that. 

I do like the name of `UnknownType`. That might be better than `_InstanceWithNoClassDefinition`

I will go with that.